lib/rails-jquery-autocomplete/autocomplete.rb

```diff
@@ -49,30 +49,36 @@ def autocomplete(object, method, options = {}, &block)
             'active_record'
           end
         end
-        define_method("get_autocomplete_order") do |method, options, model=nil|
-          method("#{get_prefix(get_object(options[:class_name] || object))}_get_autocomplete_order").call(method, options, model)
+        define_method("get_autocomplete_order") do |object, method, options, model=nil|
+          method("#{get_prefix(get_object(object))}_get_autocomplete_order").call(method, options, model)
         end
 
-        define_method("get_autocomplete_items") do |parameters|
-          method("#{get_prefix(get_object(options[:class_name] || object))}_get_autocomplete_items").call(parameters)
+        define_method("get_autocomplete_items") do |object, parameters|
+          method("#{get_prefix(get_object(object))}_get_autocomplete_items").call(parameters)
         end
 
-        define_method("autocomplete_#{object}_#{method}") do
+        # var = {:object => :method, "class_name" => "column_id", \
+        # :object => "column_id", "class_name" => :method}
+        # #Pass parameters as autocomplete :name, var
+        # :name creates the name of the action
+        define_method("autocomplete_#{name}") do
 
-          method = options[:column_name] if options.has_key?(:column_name)
+          json = Array.new
 
           term = params[:term]
 
           if term && !term.blank?
-            #allow specifying fully qualified class name for model object
-            class_name = options[:class_name] || object
-            items = get_autocomplete_items(:model => get_object(class_name), \
-              :options => options, :term => term, :method => method)
-          else
-            items = {}
+
+            object_method_hash.each do |object, method|
+              #allow specifying fully qualified class name for model object
+              #both object and method can be specified by object or id
+              items = get_autocomplete_items(object, :model => get_object(object), \
+			           :options => options, :term => term, :method => method)
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.<br>Tab detected.